### PR TITLE
GDPR-78 Sourcing list of reasons from Data Compliance API

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: prisonstaffhub-specs/src/test/groovy/uk/gov/justice/digital/hmpps/prisonstaffhub/model/TestFixture.groovy [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package uk.gov.justice.digital.hmpps.prisonstaffhub.model
 
 import geb.Browser
+import javafx.scene.chart.XYChart.Data
```
Comment (andrewrlee) [inline prisonstaffhub-specs/src/test/groovy/uk/gov/justice/digital/hmpps/prisonstaffhub/model/TestFixture.groovy]: this fella

File: backend/controllers/retentionReasons.js [hunk 2]
```diff
@@ -20,10 +20,12 @@ const retentionReasonsFactory = (elite2Api, logError) => {
       const agencies = await elite2Api.getAgencies(res.locals)
       const agency = agencies.find(a => a.agencyId === offenderDetails.agencyId).description
       const offenderUrl = getOffenderUrl(offenderNo)
+      const retentionReasons = await dataComplianceApi.getOffenderRetentionReasons(res.locals)
```
Comment (andrewrlee) [inline backend/controllers/retentionReasons.js]: would generally run those 3 queries in parallel through using `Promise.all()`.

Comment (andrewrlee) [reply backend/controllers/retentionReasons.js]:
```js
let [first, second, third] = await Promise.all([
    someThingThatReturnsAPromise(),
    otherThingThatReturnsAPromise(),
    lastThingThatReturnsAPromise()
 ]);

File: prisonstaffhub-specs/src/test/groovy/uk/gov/justice/digital/hmpps/prisonstaffhub/model/TestFixture.groovy (hunk 1 above)
Comment (brightonsbox) [reply prisonstaffhub-specs/src/test/groovy/uk/gov/justice/digital/hmpps/prisonstaffhub/model/TestFixture.groovy]: Removed

File: backend/controllers/retentionReasons.js (hunk 2 above)
Comment (brightonsbox) [reply backend/controllers/retentionReasons.js]: Cheers, done

**Pull request merged**